ghcide/src/Development/IDE/Core/Compile.hs

```diff
@@ -998,27 +998,37 @@ getDocsBatch
   :: HscEnv
   -> Module  -- ^ a moudle where the names are in scope
   -> [Name]
-  -> IO [Either String (Maybe HsDocString, Map.Map Int HsDocString)]
+  -> IO [(Name, Either String (Maybe HsDocString, Map.Map Int HsDocString))]
 getDocsBatch hsc_env _mod _names = do
     ((_warns,errs), res) <- initTc hsc_env HsSrcFile False _mod fakeSpan $ traverse findNameInfo _names
     case res of
-        Just x  -> return $ map (first $ T.unpack . showGhc) x
+        Just x  -> return $ fun x
         Nothing -> throwErrors errs
   where
+    fun :: [(Name, Either GetDocsFailure c)] -> [(Name, Either String c)]
+    fun =
+      map fun1
+     where
+      fun1 :: ((Name, Either GetDocsFailure c) -> (Name, Either String c))
+      fun1 = fmap (first $ T.unpack . showGhc)
+
     throwErrors = liftIO . throwIO . mkSrcErr
```

Comment:
I love to solve problems by composition.

Initially, I call functions `fun` (& recursive `go`) until from typesignatures I find the proper name for them.

We use additional `fmap` here, to `fmap` over `(Name,` tuple provisioning.